backend/cmmgen.ml

```diff
@@ -1275,6 +1280,20 @@ and transl_let env str (Pvalue kind : Lambda.layout) id exp transl_body =
       | Mutable, bn -> Clet_mut (v, typ_of_boxed_number bn, cexp, body)
       end
 
+and transl_let env str (layout : Lambda.layout) id exp transl_body =
+  match layout with
+  | Ptop ->
+    Misc.fatal_errorf "Variable %a with layout %a can't be compiled"
+      VP.print id Printlambda.layout layout
+  | Pbottom ->
+    let cexp = transl env exp in
+    (* N.B. [body] must still be traversed even if [exp] will never return:
+       there may be constant closures inside that need lifting out. *)
+    let _cbody = transl_body env in
```

Comment:
This should have a type annotation, just in case extra parameters are added to transl_body later...